test/SNLS_testdriver_hybrd.cc

```diff
@@ -0,0 +1,368 @@
+            r[i] = (3-2*x[i])*x[i] - x[i-1] - 2*x[i+1] + 1;
+
+         fn = (3-2*x[nDim-1])*x[nDim-1] - x[nDim-2] + 1;
+         r[nDim-1] = (1-_lambda)*fn + _lambda*(fn*fn);
+
+         if ( doComputeJ ) {
+            // F(0) = (3-2*x[0])*x[0] - 2*x[1] + 1;
+            J[SNLS_NN_INDX(0,0,nDim)] = 3 - 4*x[0];
+            J[SNLS_NN_INDX(0,1,nDim)] = -2;
+
+            // F(i) = (3-2*x[i])*x[i] - x[i-1] - 2*x[i+1] + 1;
+            for (int i=1; i<nDim-1; i++) {
+               J[SNLS_NN_INDX(i,i-1,nDim)] = -1;
+               J[SNLS_NN_INDX(i,i,nDim)]   = 3 - 4*x[i];
+               J[SNLS_NN_INDX(i,i+1,nDim)] = -2;
+            }
+
+            // F(n-1) = ((3-2*x[n-1])*x[n-1] - x[n-2] + 1)^2;
+            fn = (3-2*x[nDim-1])*x[nDim-1] - x[nDim-2] + 1;
+            double dfndxn = 3-4*x[nDim-1];
+            J[SNLS_NN_INDX(nDim-1,nDim-1,nDim)] = (1-_lambda)*(dfndxn) + _lambda*(2*dfndxn*fn);
+            J[SNLS_NN_INDX(nDim-1,nDim-2,nDim)] = (1-_lambda)*(-1) + _lambda*(-2*fn);
+         }
+
+         return true ;
+         
+      } ;
+   
+   private:
+      double _lambda ;
+      static const int _nXn = nDimSys*nDimSys ;
+};
+
+template<int nd>
+class ChebyQuad
+{
+   public:
+   static const int nDimSys = nd;
+   ChebyQuad(){}
+   ~ChebyQuad(){}
+
+   bool computeRJ(double* const r,
+                  double* const J,
+                  const double* const x)
+   {
+      const int n = nDimSys;
+      double temp1, temp2, temp, ti, d1, tk;
+      double temp3, temp4;
+      int iev;
+    
+      for (int i = 0; i < n; i++) {
+         r[i] = 0.0;
+      }
+      
+      for (int j = 0; j < n; j++) {
+         temp1 = 1.0;
+         temp2 = 2.0 * x[j] - 1.0;
+         temp = 2.0 * temp2;
+         for (int i = 0; i < n; i++) {
+               r[i] += temp2;
+               ti = temp * temp2 - temp1;
+               temp1 = temp2;
+               temp2 = ti;
+         }
+      }
+
+      tk = 1.0 / ((double) n);
+      iev = -1;
+      for (int k = 0; k < n; k++) {
+         r[k] = tk * r[k];
+         if (iev > 0) {
+               d1 = ((double) k) + 1.0;
+               r[k] += 1.0 / (d1 * d1 - 1.0);
+         }
+         iev = -iev;
+      }
+      if (J != nullptr) {
+         tk = 1.0 / ((double) n);
+         for (int j = 0; j < n; j++) {
+            temp1 = 1.0;
+            temp2 = 2.0 * x[j] - 1.0;
+            temp = 2.0 * temp2;
+            temp3 = 0.0;
+            temp4 = 2.0;
+            for (int k = 0; k < n; k++) {
+                  J[SNLS_NN_INDX(j, k, nDimSys)] = tk * temp4;
+                  ti = 4. * temp2 + temp * temp4 - temp3;
+                  temp3 = temp4;
+                  temp4 = ti;
+                  ti = temp * temp2 - temp1;
+                  temp1 = temp2;
+                  temp2 = ti;
+            }
+         }
+      }
+            
+    return true;
+   }
+
+};
+
+// This test will fail due to our delta being too aggressive which causes the solution to not move 
```

Comment:
Changes to _xiLG and _xiLO in the TrDeltaControl might allow the test to pass, depending on the nature of the "too aggressive" behavior that you reference.

Reply:
Didn't really have any luck with at least the parameters I tried :/ I'm just going to leave this in as a known failed test, and if we end up allowing classes to be templated on a delta control type class then we can work on getting this one to pass.